sympy/sets/sets.py

```diff
@@ -994,6 +993,29 @@ def _eval_imageset(self, f):
             return
         var = f.variables[0]
 
+        if expr.is_Piecewise:
+            result = S.EmptySet
+            domain_set = self
+            for i in expr.args:
+                func = i[0]
+                if i[1] is S.true:
+                    intrvl = domain_set
+                else:
+                    intrvl = i[1].as_set()
+                    intrvl = Intersection(domain_set, intrvl)
+
+                if func.is_Number:
+                    image = FiniteSet(func)
+                else:
+                    image = imageset(Lambda(var, func), intrvl)
+                result = Union(result, image)
+
+                # remove the part which has been `imaged`
+                domain_set = Complement(domain_set, intrvl)
+                if domain_set.is_EmptySet:
+                    break
```

Comment:
> Does it make any difference in the final result and are their any other advantages of doing things this way?

No, i think there would not be any difference in answer. But `without` the .is_EmptySet check it will go around the loop, as many times as the `number of pieces` in which function has been defined, (in my example 2 times without .is_EmptySet check).